tests/integration/test_design.py

```diff
@@ -3286,3 +3296,311 @@ def test_get_body_bounding_box(modeler: Modeler):
+
+
+def test_trimmed_curve(modeler: Modeler):
+    """Test Trimmed Curve class"""
+    design = modeler.create_design("trimmed_curve_edges")
+    body = design.extrude_sketch("box", Sketch().box(Point2D([0, 0]), 1, 1), 1)
+    with pytest.raises(ValueError):
+        design.bodies[0].edges[0].shape.intersect_curve(design.bodies[0].edges[1].shape)
+    # Retrieve edges and initialize TrimmedCurve objects with the gRPC client
+    edge0 = TrimmedCurve(
+        geometry=body.edges[0].shape.geometry,
+        start=body.edges[0].shape.start,
+        end=body.edges[0].shape.end,
+        interval=body.edges[0].shape.interval,
+        length=body.edges[0].shape.length,
+        grpc_client=modeler.client,  # Pass the gRPC client here
+    )
+    edge1 = TrimmedCurve(
+        geometry=body.edges[1].shape.geometry,
+        start=body.edges[1].shape.start,
+        end=body.edges[1].shape.end,
+        interval=body.edges[1].shape.interval,
+        length=body.edges[1].shape.length,
+        grpc_client=modeler.client,  # Pass the gRPC client here
+    )
+
+    edge2 = TrimmedCurve(
+        geometry=body.edges[4].shape.geometry,
+        start=body.edges[4].shape.start,
+        end=body.edges[4].shape.end,
+        interval=body.edges[4].shape.interval,
+        length=body.edges[4].shape.length,
+        grpc_client=modeler.client,  # Pass the gRPC client here
+    )
+
+    # Perform assertions and call intersect_curve
+    assert (
+        edge0.__repr__()
+        == "TrimmedCurve(geometry: <class 'ansys.geometry.core.shapes.curves.line.Line'>, "
+        "start: [-0.5 -0.5  1. ], end: [ 0.5 -0.5  1. ], "
+        "interval: Interval(start=0.0, end=1.0), length: 1.0 meter)"
+    )
+    assert edge0.length == Quantity(1, UNITS.m)
+    assert edge0.intersect_curve(edge1) == [Point3D([-0.5, -0.5, 1.0])]
+    assert edge0.intersect_curve(edge2) == []
+
+    design._update_design_inplace()
+
+
+def test_unite_interval(modeler: Modeler):
+    """Test unite interval functionality."""
+    with pytest.raises(ValueError, match="Start value must be less than end value"):
+        Interval(5, 0)
+    interval1 = Interval(0, 5)
+    interval2 = Interval(1, 3)
+
+    assert interval1.__eq__(4) == NotImplemented
+    assert not interval1.__eq__(interval2)
+    assert interval2.__eq__(interval2)
+
+    # Test unite
+    united_interval = Interval.unite(interval1, interval2)
+    assert united_interval.start == 0
+    assert united_interval.end == 5
+
+    united_interval = Interval.unite(interval2, interval1)
+    assert united_interval.start == 0
+    assert united_interval.end == 5
+
+    united_interval = Interval.unite(interval1, Interval(1, 10))
+    assert united_interval.start == 0
+    assert united_interval.end == 10
+
+    # Test self_unite
+    interval1.self_unite(interval2)
+    assert interval1.start == 0
+    assert interval1.end == 5
+
+
+def test_intersect_interval(modeler: Modeler):
```

Comment:
Remove modeler and move to test_parameterization.py (or test_shapes.py if you create it)